controllers/v2/unavailability/api.py

```diff
@@ -20,16 +21,25 @@ def __init__(self, event_repository: EventRepository = EventRepository()):
         self.event_repository = event_repository
 
     @requires_auth
-    @is_user_or_has_role(None, UserType.ROOT_ADMIN)
     def put(self, user_id, event_id):
-        args = edit_parser.parse_args()
-        success = self.event_repository.edit_event(user_id, event_id, **args)
-        if success is True:
+        @is_user_or_has_role(user_id, UserType.ROOT_ADMIN)
```

Comment:
Why move this check to here? IIRC the decorator should automatically look for a "user_id" path parameter. This kind of setup should only be necessary if the relevant user_id isn't in the path.

Reply:
Hi Ben, I changed it back. You are correct!  I just have a little bit of misunderstanding. Thanks :-)